apps/src/applab/appStorage.js

```diff
@@ -84,5 +89,95 @@ var handleReadSharedRecords = function(tableName, searchParams, callback) {
     }
     return true;
   });
-  callback(records);
+  onSuccess(records);
+};
+
+/**
+ * Updates a record in a table, accessible to all users.
+ * @param {string} record.tableName The name of the table to update.
+ * @param {string} record.id The id of the row to update.
+ * @param {Object} record Object containing other properites to update
+ *     on the record.
+ * @param {function()} onSuccess Function to call on success.
+ * @param {function(string)} onError Function to call with an error message
+ *    in case of failure.
+ */
+AppStorage.updateSharedRecord = function(record, onSuccess, onError) {
+  var tableName = record.tableName;
+  if (!tableName) {
+    onError('error updating record: missing required property "tableName"');
+    return;
+  }
+  var recordId = record.id;
+  if (!recordId) {
+    onError('error updating record: missing required property "id"');
+    return;
+  }
+  var req = new XMLHttpRequest();
+  req.onreadystatechange = handleUpdateSharedRecord.bind(req, record, onSuccess, onError);
+  var url = '/v3/apps/' + AppStorage.tempEncryptedAppId + '/shared-tables/' +
+      tableName + '/' + recordId;
+  req.open('POST', url, true);
+  req.setRequestHeader('Content-Type', 'application/json; charset=UTF-8');
+  req.send(JSON.stringify(record));
```

Comment:
OK to include tableName and id in the JSON? Seems clean that you don't need to remove them, but perhaps a little redundant since they are in the URL as well. I assume they can be validated and/or ignored on the service side.